tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rewriter/MetricProfilerListener.scala

```diff
@@ -0,0 +1,103 @@
+package at.forsyte.apalache.tla.bmcmt.rewriter
+
+import java.io.{File, FileWriter, PrintWriter}
+
+import at.forsyte.apalache.tla.bmcmt.smt.SolverContextMetrics
+import at.forsyte.apalache.tla.imp.src.SourceStore
+import at.forsyte.apalache.tla.lir.{TlaEx, UID}
+import at.forsyte.apalache.tla.lir.storage.{ChangeListener, SourceLocator}
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+  * This listener registers the SMT metrics that are created when an expression is being translated in SMT.
+  * These metrics are collected only for those expressions that have source information, which can be
+  * displayed to the user.
+  *
+  * @param sourceStore the storage of source locations
+  * @param changeListener the tracer of expression updates
+  */
+class MetricProfilerListener(sourceStore: SourceStore, changeListener: ChangeListener, outFile: File)
+    extends SymbStateRewriterListener with LazyLogging {
+  private var _metricsPerId: Map[UID, SolverContextMetrics] = Map()
```

Comment:
what is this one underscored?

Reply:
Sometimes, the underscored versions help to distinguish between the private attribute `x`, its getter (def `x`) and setter `def x_(newValue: T)...`. If you like to see more on scala getters and setter: https://stackoverflow.com/questions/26012039/scala-getters-setters-best-practice

However, it does not seem to be the case here. So it was more like a reflex to add underscore :-)